inc/options/class-wpseo-option-titles.php

```diff
@@ -215,101 +220,83 @@ public function translate_defaults() {
 	 * @return  void
 	 */
 	public function enrich_defaults() {
+		$cache_key = 'yoast_titles_rich_defaults_' . $this->option_name;
+
+		$enriched_defaults = wp_cache_get( $cache_key );
+		if ( false !== $enriched_defaults ) {
+			$this->defaults += $enriched_defaults;
+			return;
+		}
+
+		$enriched_defaults = array();
+
 		/*
 		 * Retrieve all the relevant post type and taxonomy arrays.
 		 *
 		 * WPSEO_Post_Type::get_accessible_post_types() should *not* be used here.
 		 * These are the defaults and can be prepared for any public post type.
 		 */
-		$post_type_names = get_post_types( array( 'public' => true ), 'names' );
-
-		$post_type_objects_custom = get_post_types(
-			array(
-				'public'   => true,
-				'_builtin' => false,
-			),
-			'objects'
-		);
-
-		$taxonomy_names = get_taxonomies( array( 'public' => true ), 'names' );
-
-
-		if ( $post_type_names !== array() ) {
-			foreach ( $post_type_names as $pt ) {
-				$this->defaults[ 'title-' . $pt ]              = '%%title%% %%page%% %%sep%% %%sitename%%'; // Text field.
-				$this->defaults[ 'metadesc-' . $pt ]           = ''; // Text area.
-				$this->defaults[ 'noindex-' . $pt ]            = false;
-				$this->defaults[ 'showdate-' . $pt ]           = false;
-				$this->defaults[ 'display-metabox-pt-' . $pt ] = true;
-			}
-			unset( $pt );
-		}
+		$post_type_objects = get_post_types( array( 'public' => true ), 'objects' );
 
-		if ( $post_type_objects_custom !== array() ) {
+		if ( $post_type_objects ) {
 			/* translators: %s expands to the name of a post type (plural). */
 			$archive = sprintf( __( '%s Archive', 'wordpress-seo' ), '%%pt_plural%%' );
-			foreach ( $post_type_objects_custom as $pt ) {
-				if ( ! WPSEO_Post_Type::has_archive( $pt ) ) {
-					continue;
-				}
 
-				$this->defaults[ 'title-ptarchive-' . $pt->name ]    = $archive . ' %%page%% %%sep%% %%sitename%%'; // Text field.
-				$this->defaults[ 'metadesc-ptarchive-' . $pt->name ] = ''; // Text area.
-				$this->defaults[ 'bctitle-ptarchive-' . $pt->name ]  = ''; // Text field.
-				$this->defaults[ 'noindex-ptarchive-' . $pt->name ]  = false;
+			foreach ( $post_type_objects as $pt ) {
+				$enriched_defaults[ 'title-' . $pt->name ]                   = '%%title%% %%page%% %%sep%% %%sitename%%'; // Text field.
+				$enriched_defaults[ 'metadesc-' . $pt->name ]                = ''; // Text area.
+				$enriched_defaults[ 'noindex-' . $pt->name ]                 = false;
+				$enriched_defaults[ 'showdate-' . $pt->name ]                = false;
+				$enriched_defaults[ 'display-metabox-pt-' . $pt->name ]      = true;
+				$enriched_defaults[ 'post_types-' . $pt->name . '-maintax' ] = 0; // Select box.
+
+				if ( ! $pt->_builtin ) {
+					if ( ! WPSEO_Post_Type::has_archive( $pt ) ) {
```

Comment:
To reduce complexity I would combine this line with the one above:
```php
if ( ! $pt->_builtin && WPSEO_Post_Type::has_archive( $pt ) ) {
    $enriched_defaults...
```

or 
```php
if ( $pt->_builtin || WPSEO_Post_Type::has_archive( $pt ) ) {
   continue;
}

$enriched_defaults...
```

Reply:
Thanks! Yeah, makes sense :)